src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,225 @@
+                    match=r"No cores found for the species set 'test' on the specified host."))
+        ]
+    )
+    def test_dump_genomes(self, species_list: List[str], species_set_name: str,
+                          tmp_dir: Path, expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` function
+        when connection to the server host and port can be established.
+
+        Args:
+            species_list: A list of species (genome names).
+            species_set_name: Species set (collection) name.
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                    exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            host = self.core_dbs["gallus_gallus_core_99_6"].dbc.host
+            port = self.core_dbs["gallus_gallus_core_99_6"].dbc.port
+            # user "travis" hardcoded until we find a better solution
+            orthology_benchmark.dump_genomes(species_list, species_set_name, host, port, "travis", tmp_dir)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "orth_benchmark"  # type: ignore[attr-defined]
+            for db_name in self.core_dbs:
+                assert file_cmp(Path(out_files, os.environ['USER'] + "_" + db_name + ".fasta"),
+                                Path(exp_out, db_name + ".fasta"))
+
+    def test_dump_genomes_fake_connection(self, tmp_dir: Path) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` function
+        when provided fake server connection details.
+
+        Args:
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with raises(sqlalchemy.exc.OperationalError):
+            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "default",
+                                             "fake-host", 666, "compara", tmp_dir)
+
+
+@pytest.mark.parametrize(
+    "core_names, exp_output, expectation",
+    [
+        (["mus_musculus_core_105_1", "mus_musculus_core_105_3", "mus_musculus_core_104_4"],
+         "mus_musculus_core_105_3", does_not_raise()),
+        ([], None, raises(RuntimeError,
+                          match=r"Empty list of core databases. Cannot determine the latest one."))
+    ]
+)
+def test_find_latest_core(core_names: List[str], exp_output: str, expectation: ContextManager) -> None:
+    """Tests :func:`orthology_benchmark.find_latest_core()` function.
+
+    Args:
+        core_names: A list of core database names.
+        exp_output: Expected return value of the function.
+        expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
```

Comment:
Tweak to 4-space hanging indent:
```suggestion
        expectation: Context manager for the expected exception, i.e. the test will only pass if that
            exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
```